com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/protocol/Requests.java

```diff
@@ -38,32 +38,123 @@
         public boolean supportsRunInTerminalRequest;
     }
 
-    public static class LaunchArguments extends Arguments {
+    public static class LaunchBaseArguments extends Arguments {
         public String type;
         public String name;
         public String request;
         public String projectName;
+        public String[] sourcePaths = new String[0];
+    }
+
+    public static enum CONSOLE {
+        internalConsole,
+        integratedTerminal,
+        externalTerminal;
+    }
+
+    public static class LaunchArguments extends LaunchBaseArguments {
         public String mainClass;
         public String args = "";
         public String vmArgs = "";
         public String encoding = "";
         public String[] classPaths = new String[0];
         public String[] modulePaths = new String[0];
-        public String[] sourcePaths = new String[0];
         public String cwd;
         public Map<String, String> env;
         public boolean stopOnEntry;
+        public CONSOLE console = CONSOLE.internalConsole;
     }
 
-    public static class AttachArguments extends Arguments {
-        public String type;
-        public String name;
-        public String request;
+    public static class AttachArguments extends LaunchBaseArguments {
         public String hostName;
         public int port;
         public int timeout = 30000; // Default to 30s.
-        public String[] sourcePaths = new String[0];
-        public String projectName;
+    }
+
+    public static class RunInTerminalRequestArguments extends Arguments {
+        public String kind; // Supported kind should be "integrated" or "external".
```

Comment:
kind->ternimalType

Reply:
NO, this is from the protocol definition, not our own class.